### PR TITLE
Update go to 1.21

### PR DESCRIPTION
**Pull request by orestisfl**
### Summary of your changes
Updates go to latest release, since this release should be mature enough for us.

Release notes: https://tip.golang.org/doc/go1.21

Some points of interest:

> New slices package
> 
> The new [slices](https://tip.golang.org/pkg/slices) package provides many common operations on slices, using generic functions that work with slices of any element type.
> New maps package
> 
> The new [maps](https://tip.golang.org/pkg/maps/) package provides several common operations on maps, using generic functions that work with maps of any key or element type.

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: deploy/cloudformation/config.go [hunk 1]
```diff
@@ -23,10 +23,10 @@ package main
 import (
 	"fmt"
 	"reflect"
+	"slices"
```
Comment (romulets) [inline deploy/cloudformation/config.go:26]: Nice 😄

File: vulnerability/events_creator.go [hunk 2]
```diff
@@ -31,7 +31,6 @@ import (
 	libevents "github.com/elastic/beats/v7/libbeat/beat/events"
 	"github.com/elastic/elastic-agent-libs/logp"
 	"github.com/elastic/elastic-agent-libs/mapstr"
-	"golang.org/x/exp/maps"
```
Comment (romulets) [inline vulnerability/events_creator.go:34]: Aren't you missing a `"maps"` import?

**Review by romulets — CHANGES_REQUESTED** [review]

File: vulnerability/events_creator.go (hunk 2 above)
Comment (orestisfl) [reply vulnerability/events_creator.go:34]: uhhm, yeah, no sure how this passed the pre-commit checks :sweat_smile:

**Review by romulets — APPROVED** [review]

File: .github/workflows/cloudbeat-ci.yml [hunk 3]
```diff
@@ -32,7 +32,11 @@ jobs:
           go-version-file: .go-version
 
       - name: Init Hermit
-        run: ./bin/hermit env -r >> $GITHUB_ENV
+        run: |
+          # Pre-fetch hermit binaries otherwise pre-commit can fail when running multiple commands in parallel. This
+          # leads to hermit attempting to download the same binary multiple times which results in a race condition.
```
Comment (moukoublen) [inline .github/workflows/cloudbeat-ci.yml:37]: Super optional, but should we put the comments as yaml comments before run and not as shell comments inside of it?

**Review by moukoublen — APPROVED** [review]

**Pull request merged**